generator/src/main/java/com/graphicsfuzz/generator/mutateapi/CompositeValue.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.graphicsfuzz.generator.mutateapi;
+
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.expr.TypeConstructorExpr;
+import com.graphicsfuzz.common.ast.type.BasicType;
+import com.graphicsfuzz.common.ast.type.Type;
+import com.graphicsfuzz.generator.semanticschanging.LiteralFuzzer;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class CompositeValue implements Value {
+  private final Type type;
+  private final Optional<List<Value>> valueList;
+
+  public CompositeValue(Type type, Optional<List<Value>> valueList) {
+    this.type = type;
+    this.valueList = valueList;
+  }
+
+  @Override
+  public Type getType() {
+    return type;
+  }
+
+  @Override
+  public boolean valueIsKnown() {
+    return valueList.isPresent();
+  }
+
+  @Override
```

Comment:
The equals methods look incomplete.

Please adjust as suggested in BooleanValue to have one equals method that overrides the one from Object, and get rid of the "boolean equals(CompositeValue that)" overload.

Then the logic should be easy: just need to return whether the types are .equals and the values are .equals.